frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.effects.ts

```diff
@@ -177,19 +179,79 @@ export class FeatureFlagsEffects {
 
   fetchFeatureFlagById$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(FeatureFlagsActions.actionFetchFeatureFlagById),
+      ofType(featureFlagsActions.actionFetchFeatureFlagById),
       map((action) => action.featureFlagId),
       filter((featureFlagId) => !!featureFlagId),
       switchMap((featureFlagId) =>
         this.featureFlagsDataService.fetchFeatureFlagById(featureFlagId).pipe(
           map((data: FeatureFlag) => {
-            return FeatureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
+            return featureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
           }),
-          catchError(() => [FeatureFlagsActions.actionFetchFeatureFlagByIdFailure()])
+          catchError(() => [featureFlagsActions.actionFetchFeatureFlagByIdFailure()])
         )
       )
     )
   );
 
+  emailFeatureFlagData$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(featureFlagsActions.actionEmailFeatureFlagData),
+      map((action) => ({ featureFlagId: action.featureFlagId })),
+      withLatestFrom(this.store$.pipe(select(selectCurrentUser))),
+      filter(([{ featureFlagId }, { email }]) => !!featureFlagId && !!email),
+      switchMap(([{ featureFlagId }, { email }]) =>
+        this.featureFlagsDataService.emailFeatureFlagData(featureFlagId, email).pipe(
+          map(() => {
+            this.notificationService.showSuccess(`Email will be sent to ${email}`);
+            return featureFlagsActions.actionEmailFeatureFlagDataSuccess();
+          }),
+          catchError(() => [featureFlagsActions.actionEmailFeatureFlagDataFailure()])
+        )
+      )
+    )
+  );
+
+  exportFeatureFlagsDesign$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(featureFlagsActions.actionExportFeatureFlagDesign),
+      map((action) => ({ featureFlagIds: action.featureFlagIds })),
+      filter(({ featureFlagIds }) => !!featureFlagIds),
+      switchMap(({ featureFlagIds }) =>
+        this.featureFlagsDataService.exportFeatureFlagsDesign(featureFlagIds).pipe(
+          map((data: FeatureFlag[]) => {
```

Comment:
the data-processing logic tucked into this effect would be excellently placed in it's own service so we can keep the effects file clear of stuff it doesn't need to know about. same goes for the private "download" function, DOM stuff shouldn't be in effects file. it's painful to see this stuff copy-pasted multiple times. how about an 'export-helper.service' or something?